source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -297,13 +313,34 @@ Http::Code HystrixSink::handlerHystrixEventStream(absl::string_view,
   return Http::Code::OK;
 }
 
-void HystrixSink::flush(Stats::Source&) {
+void HystrixSink::flush(Stats::Source& source) {
   if (callbacks_list_.empty()) {
     return;
   }
   incCounter();
   std::stringstream ss;
   Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+
+  // Save a map of the relevant histograms per cluster in a convenient format.
+  std::unordered_map<std::string, QuantileLatencyMap> time_histograms;
+  for (const Stats::ParentHistogramSharedPtr histogram : source.cachedHistograms()) {
+    // histogram->name() on clusters of the format "cluster.cluster_name.histogram_name"
+    // i.e. "cluster.service1.upstream_rq_time".
+    const std::vector<std::string> split_name = absl::StrSplit(histogram->name(), '.');
+    if (split_name.size() >= 2 && split_name[0] == "cluster" &&
```

Comment:
I just realized that you're essentially repeating work already done by Envoy.  Rather than manually parsing the string, you could check the tag and tag_extracted_name since Envoy parses those for you. It may be a bit more code, but I think it's a bit safer since the regexes are a little more specific than splitting on `.`. It's up to you if you'd rather use it or not, but I figured I'd present the alternative. For example (this may not compile since I haven't tried, but the rough idea is there :) ):
```
if (histogram->tag_extracted_name() == "cluster.upstream_rq_time") {
  // TODO(mrice32): add an Envoy utility function to look up and return a tag for a metric.
  auto it = std::find_if(histogram->tags().begin(), histogram->tags().end(), [](const Stats::Tag& tag) {
    return tag.name_ == Config::TagNames::get().CLUSTER_NAME;
  });
  // Should always be true, so we probably want to log or something if not.
  if (it == histogram.tags().end()) {
    // We didn't find the cluster name despite finding the correct stat.
  } else {
    QuantileLatencyMap& hist_map = time_histograms[it->value_];
    ...
  }
}

Reply:
I don't know if we should log if a cluster_name tag is not found. What could be the reason? Anything the user can do about it? If not, I think we can just verify it is not histogram.tags().end().

Reply:
I think it would imply that something is broken since we expect that this should never happen.  I wouldn't say it's a catastrophic failure, so it's not a crashable event, but logging an error to the console might catch someone's attention so that they could file a bug against Envoy.  I'm not super opinionated, so it's ultimately up to you.

Reply:
IMO this is what ASSERT is for. It should stop tests from passing so you can investigate. I suspect a log would just slow things down and no one would notice.

Reply:
+1 for ASSERT